### PR TITLE
Fix incorrect default behaviour

### PR DESCRIPTION
**Pull request by Mahmood-Darwish**
This PR fixes Scheduler track not being visible in e2e because default behavior is set incorrectly and it tries to use an unfinished feature. More details in the bug report.

Bug: http://b/243522811

### CODE REVIEW

**Review by florian-kuebler — COMMENTED** [review]
Thanks for fixing this. However, I think we also need to do this at on_actionCaptureOptions_triggered

File: src/OrbitQt/orbitmainwindow.cpp [hunk 1]
```diff
@@ -1166,18 +1166,25 @@ void OrbitMainWindow::LoadCaptureOptionsIntoApp() {
...
+    bool const collect_callstack_on_thread_state_change =
+        settings
+            .value(
+                kEnableCallStackCollectionOnThreadStateChanges,
+                orbit_qt::CaptureOptionsDialog::kThreadStateChangeCallStackCollectionDefaultValue)
+            .toBool();
+    if (settings.value(kCollectThreadStatesSettingKey, false).toBool()) {
+      if (!collect_callstack_on_thread_state_change) {
+        app_->SetThreadStateChangeCallstackCollection(
+            CaptureOptions::kNoThreadStateChangeCallStackCollection);
```
Comment (florian-kuebler) [inline src/OrbitQt/orbitmainwindow.cpp:1180]: Offtopic question: Why is this method not simply accepting a bool?

Comment (Mahmood-Darwish) [reply src/OrbitQt/orbitmainwindow.cpp:1180]: Shouldn't we have an unspecified option? That's why we use an enum instead of a bool.

**Comment by Mahmood-Darwish** [comment]
@florian-kuebler Fixed the second part of the bug. This should work now right?

File: src/OrbitQt/orbitmainwindow.cpp (hunk 1 above)
Comment (florian-kuebler) [reply src/OrbitQt/orbitmainwindow.cpp:1180]: Right. So this is for metrics collections. Fine for me.

**Review by florian-kuebler — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**